torchaudio/__init__.py

```diff
@@ -242,6 +280,11 @@ def sox_signalinfo_t():
         >>> si.precision = 16
         >>> si.length = 0
     """
+
+    if get_audio_backend() != "sox":
```

Comment:
You could create a generic decorator to write this less often

```
def _backend_guard(backends):
    def decorator(fn):
        @functools.docs-decorator(fn) # not sure about the name
        def _fn(*args, **kwargs):
            if get_audio_backend() not in backends:
                raise Runtime("fn {} requires backend to be one of".format(fn.__name__, backends)
           fn(*args, **kwargs)
       return _fn

```


```
@backend_support(['sox'])
def sox_signalinfo_t():
      ....
```

Reply:
Nice, but let's make that a separate PR.

Reply:
Alright, added to standardize import error. :)